docs/language_specification/statements/reset_instruction.md

```diff
@@ -0,0 +1,67 @@
+A **`reset`** instruction performs a measurement plus a Pauli X on the qubit argument.
+The general form of a reset instruction is as follows:
```

Comment:
Specify what a reset does at a higher level? i.e it places the qubit to a ground or 0 state. This is done by performing a measurement and a potential Pauli X on the qubit argument.

Reply:
Also a reset does not always need to have a Pauli X

Reply:
@juanboschero Thanks for the comments! Can you please comment as well on the [Confluence](https://qutech-sd.atlassian.net/wiki/spaces/QW/pages/3523051530/Research+reset+instruction+semantic) page dedicated to the investigation of the `reset` instruction?

I think the fact that we don't talk about a conditional Pauli X here is because we don't have yet conditional operations on the language.

Reply:
In the spec we need to describe what from a cQASM point of view is the meaning of the `reset` instruction. Ultimately, it resets the state of the qubit to the |0> state (in the Z-basis). Then we choose to define further what according to us this means through the implementation definition that IBM uses, _i.e._, the qubit is measured and based on the measurement outcome a Pauli-X is applied when 1, and nothing is done when 0, ultimately resulting in a qubit in the |0> state.

So if backends support the `reset` instruction natively, then they need to implement it accordingly. If they instead reset the state of the qubit in another way, they need to make sure that the result is the same as measuring it and performing a conditional Pauli-X gate, otherwise it doesn't perform a `reset` instruction as cQASM users expect it.

So it is important that the 'meaning' of the `reset` instruction is given in the spec.

Note that, because we currently do not support conditional statements in the language (yet), only backends that support the `reset` instruction natively can make use of it, _i.e._, we cannot yet decompose the `reset` instruction in OpenSquirrel, as the language is not rich enough, atm.

Reply:
OK, can you please then add a suggestion for the `reset` instruction text?

But notice that these are the current texts for both `measure` and `reset` instructions:

> A **`measure`** instruction performs a measurement to its qubit argument and
assigns the outcome to a bit variable.
> A **`reset`** instruction performs a measurement plus a Pauli X on the qubit argument.